src/reanimated2/NativeMethods.ts

```diff
@@ -21,36 +19,42 @@ export interface MeasuredDimensions {
   pageY: number;
 }
 
-const isNativeIndefined = shouldBeUseWeb();
+const isNativeUndefined = shouldBeUseWeb();
 
 export function measure(
   animatedRef: RefObjectFunction<Component>
-): MeasuredDimensions {
+): MeasuredDimensions | null {
   'worklet';
-  if (!_WORKLET || isNativeIndefined) {
+  if (!_WORKLET || isNativeUndefined) {
     console.warn(
-      '[reanimated.measure] method cannot be used for web or Chrome Debugger'
+      '[Reanimated] measure() cannot be used for web or Chrome Debugger'
     );
-    return {
-      x: NaN,
-      y: NaN,
-      width: NaN,
-      height: NaN,
-      pageX: NaN,
-      pageY: NaN,
-    };
+    return null;
   }
   const viewTag = animatedRef();
-  const result = _measure(viewTag);
-  if (result.x === -1234567) {
-    throw new Error(`The view with tag ${viewTag} could not be measured`);
-  }
-  if (isNaN(result.x)) {
+  if (viewTag === -1) {
     console.warn(
-      'Trying to measure a component which gets view-flattened on Android. To disable view-flattening, set `collapsable={false}` on this component.'
+      `[Reanimated] The view with tag ${viewTag} is not a valid argument for measure()`
     );
+    return null;
+  }
+  try {
+    const result = _measure(viewTag);
+    if (result.x === -1234567) {
+      throw new Error(
+        `The view with tag ${viewTag} returned an invalid measurement response`
+      );
+    }
+    if (isNaN(result.x)) {
+      console.warn(
+        '[Reanimated] Trying to measure a component which gets view-flattened on Android. To disable view-flattening, set `collapsable={false}` on this component.'
+      );
+    }
+    return result;
+  } catch (e) {
```

Comment:
What exception is supposed to be caught here?

Reply:
Aside from catching the manual throw described [here](https://github.com/software-mansion/react-native-reanimated/pull/3453/files#r961108367), this is supposed to catch an error caused by measuring any invisible view, like the offscreen `FlatList` item described in the docs.

I suspect (but I'm not confident) that the `viewTag === -1` check above basically handles this, but since one of the main purposes of this PR is to avoid letting `measure()` crash the parent app, it seemed safer to keep it wrapped in a `try`.

Thoughts?

Reply:
> the main purposes of this PR is to avoid letting measure() crash the parent app

Yeah, that's the main goal 😄 

I think we would like to avoid using `try`/`catch` blocks if possible, i.e. use `console.warn` instead of `throw new Error()` in JS code.

Reply:
I've just checked the native part of the code and it looks like the native side shouldn't throw any exceptions. If some error occurs (e.g. view has not been rendered yet or it has been flattened), it returns `NaN` (which is still better than random values, see https://github.com/software-mansion/react-native-reanimated/pull/3414), or `-1234567` (which I would rather not comment). So I think we can remove the `try`/`catch` block here.